### PR TITLE
feat: support color names and hex colors for better auto complete

### PR DESCRIPTION
**Pull request by domoritz**
Screen Shot 2019-07-13 at 1 19 18 PM

### CODE REVIEW

**Comment by domoritz** [comment]
Yep, we can add color names. We still need string, though, because of `rgb(0,0,0)`.

**Comment by domoritz** [comment]
We can get all colors from https://developer.mozilla.org/en-US/docs/Web/CSS/color_value with this query `[...document.querySelectorAll('#colors_table code')].map(e => e.innerText).filter(e => !e.startsWith('#'))`.

**Comment by domoritz** [comment]
Screen Shot 2019-07-15 at 9 54 43 AM

File: src/vega.schema.ts [hunk 1]
```diff
@@ -33,7 +33,163 @@ export {VgSortField, VgUnionSortField, VgCompare, VgTitle, LayoutAlign, Projecti
...
+  | 'wheat'
+  | 'whitesmoke'
+  | 'yellowgreen'
+  | 'rebeccapurple';
+
+export type Color =
+  | ColorName
+  | HexColor
+  // RGB, HSL, HSV, HWB, CMYK colors
+  | string;
```
Comment (kanitw) [inline src/vega.schema.ts]: should we add Regex pattern for string now that it's just `rgb(...)` / rgba  etc. at this point?

Comment (domoritz) [reply src/vega.schema.ts]: I don't think it's worth it (it'll take me a while to get the regex right) at this point as it won't improve autocomplete.

Comment (domoritz) [reply src/vega.schema.ts]: The syntax gets pretty subtle so I don't really want to dive into it. See https://developer.mozilla.org/en-US/docs/Web/CSS/color_value.

**Comment by kanitw** [comment]
Okie :)

**Pull request merged**